### PR TITLE
[Qt] Add simple opt-in-RBF support

### PR DESCRIPTION
**Pull request by jonasschnelli**
**1) Adds an option to set the opt-in-RBF flag when creating a transaction**
bildschirmfoto 2016-06-09 um 11 14 44

**2) Adds a fee bump option to the context menu.**
The option is only enabled if the transaction `isFromMe` and signaling RBF.

bildschirmfoto 2016-06-09 um 11 12 06

bildschirmfoto 2016-06-09 um 11 10 58

For the full "RBF experience" we also should consider:
https://github.com/bitcoin/bitcoin/pull/7817 (attribute replaceable (RBF) transactions)
https://github.com/bitcoin/bitcoin/pull/7826 (show conflicts of unconfirmed transactions in the UI)

### CODE REVIEW

File: src/qt/forms/sendcoinsdialog.ui [hunk 1]
```diff
@@ -1257,6 +1257,16 @@
       </widget>
      </item>
      <item>
+      <widget class="QCheckBox" name="optInRBF">
+       <property name="text">
+        <string>Signal Replaceability</string>
+       </property>
+       <property name="toolTip">
+        <string>Signals that this transaction can be replaced with a transation paying higher fees (as long as the transaction is NOT confirmed).</string>
```
Comment (paveljanik) [inline src/qt/forms/sendcoinsdialog.ui]: transation -> transaction

File: src/wallet/wallet.cpp [hunk 2]
```diff
@@ -1981,7 +1982,122 @@ bool CWallet::SelectCoins(const vector<COutput>& vAvailableCoins, const CAmount&
     return res;
 }
 
-bool CWallet::FundTransaction(CMutableTransaction& tx, CAmount& nFeeRet, bool overrideEstimatedFeeRate, const CFeeRate& specificFeeRate, int& nChangePosInOut, std::string& strFailReason, bool includeWatching, bool lockUnspents, const CTxDestination& destChange)
+bool CWallet::BumpFee(const CWalletTx& wtxIn, CWalletTx& wtxOut, CReserveKey& reservekeyOut, CAmount& nOldFee, CAmount& newFee, unsigned int blocksToConfirm)
+{
+    // ownly bump the fee if the transation signals opt-in-RBF after BIP125
```
Comment (paveljanik) [inline src/wallet/wallet.cpp]: only

**Comment by luke-jr** [comment]
"signals replaceability" doesn't really seem useful terminology for users. Maybe "Allow increasing fee later" or something would make more sense.

IMO there should be some way for the user to actually control what the fee is bumped to.

**Comment by laanwj** [comment]
Unfortunately this missed the 0.13 feature merge window. We should merge it soon after 0.13 is branched off.

**Comment by btcdrak** [comment]
needs rebase

**Comment by rebroad** [comment]
is anyone working on rebasing this or shall I?

**Comment by jonasschnelli** [comment]
I'm currently not working on this and I have no near future plans to work on this.
We should first review and merge https://github.com/bitcoin/bitcoin/pull/8456.

**Comment by jonasschnelli** [comment]
Closing. We need RPC support first.

**Pull request closed without merging**